test/test_parabolic_2d.jl

```diff
@@ -52,20 +52,22 @@ isdir(outdir) && rm(outdir, recursive = true)
 
     t = 0.0
     # pass in `boundary_condition_periodic` to skip boundary flux/integral evaluation
-    Trixi.calc_gradient!(gradients, ode.u0, t, mesh, equations_parabolic,
+    Trixi.calc_gradient!(gradients, Base.parent(ode.u0), t, mesh, equations_parabolic,
                          boundary_condition_periodic, dg, cache, cache_parabolic)
     @unpack x, y, xq, yq = mesh.md
     @test getindex.(gradients[1], 1) ≈ 2 * xq .* yq
     @test getindex.(gradients[2], 1) ≈ xq .^ 2
 
     u_flux = similar.(gradients)
-    Trixi.calc_viscous_fluxes!(u_flux, ode.u0, gradients, mesh, equations_parabolic,
+    Trixi.calc_viscous_fluxes!(u_flux, Base.parent(ode.u0), gradients, mesh,
+                               equations_parabolic,
                                dg, cache, cache_parabolic)
     @test u_flux[1] ≈ gradients[1]
     @test u_flux[2] ≈ gradients[2]
 
-    du = similar(ode.u0)
-    Trixi.calc_divergence!(du, ode.u0, t, u_flux, mesh, equations_parabolic,
+    du = similar(Base.parent(ode.u0))
+    Trixi.calc_divergence!(du, Base.parent(ode.u0), t, u_flux, mesh,
```

Comment:
Same question as for `calc_gradient!` (maybe dispatching on the first two arguments).